File: Nx3CS/bin/test/src/nx3/PScoreSystemsGenerator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PScoreSystemsGenerator : global::haxe.lang.HxObject {
		public    PScoreSystemsGenerator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PScoreSystemsGenerator(global::Array<object> bars){
			unchecked {
				#line 15 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
				global::nx3.PScoreSystemsGenerator.__hx_ctor_nx3_PScoreSystemsGenerator(this, bars);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PScoreSystemsGenerator(global::nx3.PScoreSystemsGenerator __temp_me115, global::Array<object> bars){
			unchecked {
				#line 17 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
				__temp_me115.bars = bars;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
				return new global::nx3.PScoreSystemsGenerator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
				return new global::nx3.PScoreSystemsGenerator(((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (arr[0]) ))) ));
			}
			#line default
		}
		
		
		public  global::Array<object> bars;
		
		public virtual   global::Array<object> getsSystems(global::Array<double> systemwidths){
			unchecked {
				#line 23 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
				global::Array<object> tempbars = this.bars.copy();
				global::Array<object> result = new global::Array<object>();
				int sysidx = 0;
				object prevbarAttributes = default(object);
				while (( tempbars.length > 0 )){
					#line 29 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
					double syswidth = default(double);
					#line 29 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
					if ((( (( ( sysidx < 0 ) || ( sysidx > ( systemwidths.length - 1 ) ) )) ? (default(global::haxe.lang.Null<double>)) : (new global::haxe.lang.Null<double>(systemwidths[sysidx], true)) )).hasValue) {
						#line 29 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
						syswidth = systemwidths[sysidx];
					}
					 else {
						#line 29 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
						syswidth = systemwidths[0];
					}
					
					#line 30 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
					global::nx3.PSystemGenerator generator = new global::nx3.PSystemGenerator(((global::Array<object>) (tempbars) ), ((object) (new global::haxe.lang.DynamicObject(new global::Array<int>(new int[]{480140652, 1756293853, 1944669088}), new global::Array<object>(new object[]{true, true, ( sysidx == 0 )}), new global::Array<int>(new int[]{}), new global::Array<double>(new double[]{}))) ), ((object) (prevbarAttributes) ), ((double) (syswidth) ), ((global::nx3.IBarWidthCalculator) (new global::nx3.PBarWidthCalculator()) ));
					global::nx3.PSystem system = generator.getSystem();
					prevbarAttributes = system.getLastBarAttributes();
					result.push(system);
					sysidx++;
				}
				
				#line 37 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
				return result;
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
				switch (hash){
					case 1091626816:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
						this.bars = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (@value) ))) );
						#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
				switch (hash){
					case 1587408135:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getsSystems") ), ((int) (1587408135) ))) );
					}
					
					
					case 1091626816:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
						return this.bars;
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
				switch (hash){
					case 1587408135:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
						return this.getsSystems(((global::Array<double>) (global::Array<object>.__hx_cast<double>(((global::Array) (dynargs[0]) ))) ));
					}
					
					
					default:
					{
						#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
				baseArr.push("bars");
				#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
				{
					#line 11 "F:\\nx3\\src\\nx3\\PScoreSystemsGenerator.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Nx3Java/java/src/nx3/PScoreSystemsGenerator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PScoreSystemsGenerator extends haxe.lang.HxObject
{
	public    PScoreSystemsGenerator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PScoreSystemsGenerator(haxe.root.Array<nx3.PBar> bars)
	{
		nx3.PScoreSystemsGenerator.__hx_ctor_nx3_PScoreSystemsGenerator(this, bars);
	}
	
	
	public static   void __hx_ctor_nx3_PScoreSystemsGenerator(nx3.PScoreSystemsGenerator __temp_me132, haxe.root.Array<nx3.PBar> bars)
	{
		__temp_me132.bars = bars;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PScoreSystemsGenerator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PScoreSystemsGenerator(((haxe.root.Array<nx3.PBar>) (arr.__get(0)) ));
	}
	
	
	public  haxe.root.Array<nx3.PBar> bars;
	
	public   haxe.root.Array<nx3.PSystem> getsSystems(haxe.root.Array<java.lang.Object> systemwidths)
	{
		haxe.root.Array<nx3.PBar> tempbars = this.bars.copy();
		haxe.root.Array<nx3.PSystem> result = new haxe.root.Array<nx3.PSystem>();
		int sysidx = 0;
		java.lang.Object prevbarAttributes = null;
		while (( tempbars.length > 0 ))
		{
			double syswidth = 0.0;
			if (( ! (( (( (( ( sysidx < 0 ) || ( sysidx > ( systemwidths.length - 1 ) ) )) ? (null) : (((double) (haxe.lang.Runtime.toDouble(systemwidths.__get(sysidx))) )) )) == null )) )) 
			{
				syswidth = ((double) (haxe.lang.Runtime.toDouble(systemwidths.__get(sysidx))) );
			}
			 else 
			{
				syswidth = ((double) (haxe.lang.Runtime.toDouble(systemwidths.__get(0))) );
			}
			
			nx3.PSystemBarsGenerator generator = new nx3.PSystemBarsGenerator(((haxe.root.Array<nx3.PBar>) (tempbars) ), ((java.lang.Object) (new haxe.lang.DynamicObject(new haxe.root.Array<java.lang.String>(new java.lang.String[]{"showFirstClef", "showFirstKey", "showFirstTime"}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{true, true, ( sysidx == 0 )}), new haxe.root.Array<java.lang.String>(new java.lang.String[]{}), new haxe.root.Array<java.lang.Object>(new java.lang.Object[]{}))) ), ((java.lang.Object) (prevbarAttributes) ), ((double) (syswidth) ), ((nx3.IBarWidthCalculator) (new nx3.PBarWidthCalculator()) ));
			nx3.PSystem system = generator.getSystem();
			prevbarAttributes = system.getLastBarAttributes();
			result.push(system);
			sysidx++;
		}
		
		return result;
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef755 = true;
			switch (field.hashCode())
			{
				case 3016384:
				{
					if (field.equals("bars")) 
					{
						__temp_executeDef755 = false;
						this.bars = ((haxe.root.Array<nx3.PBar>) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef755) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef756 = true;
			switch (field.hashCode())
			{
				case 500264327:
				{
					if (field.equals("getsSystems")) 
					{
						__temp_executeDef756 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getsSystems"))) );
					}
					
					break;
				}
				
				
				case 3016384:
				{
					if (field.equals("bars")) 
					{
						__temp_executeDef756 = false;
						return this.bars;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef756) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef757 = true;
			switch (field.hashCode())
			{
				case 500264327:
				{
					if (field.equals("getsSystems")) 
					{
						__temp_executeDef757 = false;
						return this.getsSystems(((haxe.root.Array<java.lang.Object>) (dynargs.__get(0)) ));
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef757) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("bars");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


